reprounzip-singularity/Singularity_unpacker.py

```diff
@@ -0,0 +1,236 @@
+import copy
+import tarfile
+import os
+import sys
+import yaml
+import subprocess
+
+# Open outer tar, the RPZ file
+SINGULARITY_DIR = "../../.singularity.d"
+RUN_ENV_FILE="90-environment.sh"
+ENV_DIR = SINGULARITY_DIR+"/env"
+OVERLAY_IMAGE = "repro_overlay.img"
+IMAGE_TAR_FILE = "new.tar.gz"
+MOUNT_DIR = "/mnt/"
+source_config="METADATA/config.yml"
+APP_DIR = "../../app_structure/sample_app"
+SCIF_DIR = "../../app_structure/scif"
+APP_ENV_DIR = APP_DIR+"/scif/env"
+
+
+def write_env_file(env,env_file,apps=False):
+    if apps:
+        env_file = os.path.join(APP_ENV_DIR, env_file)
+    else:
+        env_file = os.path.join(ENV_DIR, env_file)
+    with open(env_file, 'w+') as f:
+        for key,value in env.items():
+            f.write(key+"='"+value+"'\n")
+    bashCommand = "chmod +x {}".format(env_file)
+    process = subprocess.Popen(bashCommand.split(), stdout=subprocess.PIPE)
+    output, error = process.communicate()
+
+  
+
+def make_runscript(workingdir, binary, run_file, apps=False):
+    cmd = "cd {0}\n{1} {2}".format(workingdir, binary, run_file)
+    run_cmd = "run()\n{\n echo \"Nothing to run. Run Apps!!\"\n}\n\n"
+    if apps:
+        with open(os.path.join(APP_DIR+"/scif","runscript"), 'w') as f:
+            f.write(cmd)
+    else:
+        run_cmd = "run()\n"
+        run_cmd += "{\n\t" + cmd + "\n}\n"
+    upload_download_cmd = "upload()\n{\n"
+    upload_download_cmd += "\tcp " + MOUNT_DIR + "\"$1\" " + workingdir + "\n}\n\n"
+    upload_download_cmd += "download()\n{\n"
+    upload_download_cmd += "\tcp {0}/".format(workingdir) + "\"$1\" " + MOUNT_DIR + "\n}\n\n"
+    upload_download_cmd += "if [ \"$#\" -gt 0 ]\nthen \n \tcase \"$1\" in \n\t\t\"upload\"|\"download\")\n\t\t\t\"$1\" \"$2\" \n\t\t;;\n\t\t*)\n\t\techo \"Invalid arguments!!\" >&2\n\t\texit 1\n\t\t;;\n\tesac\nelse\n\trun\nfi"
+    with open(os.path.join(SINGULARITY_DIR, "runscript"), 'w') as f:
+	f.write(run_cmd)
+        f.write(upload_download_cmd)
+
+def make_main_app_base_script(app_name):
```

Comment:
You might need to "sanitize" the `app_name` since it might not be suitable for a filename or (unquoted) variable name :wink: People might get very descriptive now that they'll have Binal's GUI.